datahub/main.py

```diff
@@ -71,29 +71,21 @@ def create_opal_data(data: OpalDictData | OpalArrayData) -> dict[str, float]:
     Returns:
         A Dict of the Opal data that has just been added to the Dataframe
     """
-    dict_data = dict(data)
-    raw_data = {}
-
-    array_keys = list(OpalDictData.__fields__.keys())
-    array_keys[5:5] = ["na", "na", "na"]
-
-    if "array" in dict_data:
-        if not len(dict_data["array"]) == 45:
-            raise HTTPException(
-                status_code=400, detail="Array has invalid length. Expecting 45 items."
-            )
-        for x in range(45):
-            if not array_keys[x] == "na":
-                raw_data[array_keys[x]] = dict_data["array"][x]
-    else:
-        raw_data = dict_data
+    raw_data = data.dict()
 
     # TODO: Change print statements to more formal logging
     print(dt.opal_df)
+
+    if isinstance(data, OpalArrayData) and not len(raw_data["array"]) == 45:
+        raise HTTPException(
+            status_code=400, detail="Array has invalid length. Expecting 45 items."
+        )
+
     dt.opal_df.opal.append(raw_data)
```

Comment:
Keeping this in line with my suggestions about what `.append()` should take as inputs, I think we should use `raw_data["array"]` as the input when it's an array.